### PR TITLE
[SPARK-43933][SQL][PYTHON][CONNECT] Add linear regression aggregate functions to Scala and Python

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
Based @HyukjinKwon 's suggestion, this PR want add linear regression aggregate functions to Scala and Python API.
These functions show below.

- `regr_avgx`

- `regr_avgy`

- `regr_count`

- `regr_intercept`

- `regr_r2`

- `regr_slope`

- `regr_sxx`

- `regr_sxy`

- `regr_syy`

### Why are the changes needed?
Add linear regression aggregate functions to Scala and Python API

### Does this PR introduce _any_ user-facing change?
'No'.
New feature.

### How was this patch tested?
New test cases.

### CODE REVIEW

**Comment by amaliujia** [comment]
LGTM

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 1]
```diff
@@ -1043,6 +1043,171 @@ object functions {
...
+  def regr_avgx(y: Column, x: Column): Column = withAggregateFunction { RegrAvgX(y.expr, x.expr) }
+
+  /**
+   * Aggregate function: returns the average of the independent variable for non-null pairs
+   * in a group, where `y` is the dependent variable and `x` is the independent variable.
+   *
+   * @group agg_funcs
+   * @since 3.5.0
+   */
+  def regr_avgx(y: String, x: String): Column = regr_avgx(Column(y), Column(x))
```
Comment (zhengruifeng) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala:1062]: @beliefer I think we don't want to support column name on Scala side.

Comment (amaliujia) [reply sql/core/src/main/scala/org/apache/spark/sql/functions.scala:1062]: @zhengruifeng could you clarify it? Do we usually only care string version in Python side?

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala [hunk 2]
```diff
@@ -463,6 +463,36 @@ class DataFrameAggregateSuite extends QueryTest
     checkAggregatesWithTol(sparkKurtosis, Row(-1.5), absTol)
   }
 
+  test("linear regression") {
+
+    val sparkRegrAvgX = testData2.agg(regr_avgx($"a", $"b"))
+    checkAnswer(sparkRegrAvgX, Row(1.5))
```
Comment (zhengruifeng) [inline sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala:469]: can we compare the result with the sql side? the numbers are hard to check

```
checkAnswer(testData2.agg(regr_avgx($"a", $"b")), testData2.selectExpr('regr_avgx(a, b)'))
```

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala (hunk 1 above)
Comment (zhengruifeng) [reply sql/core/src/main/scala/org/apache/spark/sql/functions.scala:1062]: for new functions, we should follow https://github.com/apache/spark/blob/56038b5c87508d2211a41b47c192598e42d6c121/sql/core/src/main/scala/org/apache/spark/sql/functions.scala#L59-L61

**Review by zhengruifeng — APPROVED** [review]

**Comment by zhengruifeng** [comment]
merged to master

**Comment by beliefer** [comment]
@zhengruifeng @amaliujia Thank you!

**Pull request closed without merging**